### DISCUSSION TITLE
Restore from alternate storage location

### DISCUSSION THREAD

**Discussion by tiberiomorris** [vmware-tanzu/velero#5600] [Community support Q&A] [answered]
I set up a kubernetes cluster with its own default storage location but also want to use an application backup from another cluster's storage location (s3 bucket) to restore into this new cluster. In other words, perform a cluster-to-cluster application migration. I was able to add the storage location using the **backup-location create** command and omitting the default flag but how do I access the backups in this bucket from the cluster if it is not set as default? I thought that the restore command would have a flag that allows you to point to another storage class, rather than automatically go with the default, but I was unable to find one with the help command or through online searches.

Thank you!

**Comment by sseago** [comment, marked as answer]
If you have a backup which uses a different backup storage location, that should be referenced in the backup -- so when the restore looks to the backup, it will use the BSL from that backup. When you add a new BSL to your cluster, velero will sync its backups to the current cluster, so all of the backups from that storage location should show up in your cluster pointing to the appropriate BSL. You should just be able to tell Velero which backup to restore, and it should find it. At what point is this failing for you?

**Comment by tiberiomorris** [comment]
I understand now. And you are correct, when I ran the restore it found the backup. A new problem has arisen though and under the Phase column, this BSL is showing as Unavailable. So the restore backup executed but it did not execute successfully. Perhaps there was an issue with the way that I attempted to add this BSL to this other cluster.
